app/src/main/java/com/example/myapplication/vm/LoginViewModel.kt

```diff
@@ -0,0 +1,46 @@
+package com.example.myapplication.vm
+
+import android.app.Application
+import android.content.Context
+import android.content.SharedPreferences
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.MutableLiveData
+
+class LoginViewModel (app: Application): AndroidViewModel(app) {
+
+    val email = MutableLiveData<String?>()
+    val password = MutableLiveData<String?>()
+    val toast = MutableLiveData<Boolean>()
```

Comment:
Finding names for things sometimes is quite a task. A good way to name things is to think in what it does. Here you don't have a toast, you have something more specific. Later the fragment will use _that_ to show a toast.

Reply:
I agree with the Lisan's comment. VM shouldn't know about view components, so, toast isn't the best name for this case.